code/messaging/src/phone/java/com/adobe/marketing/mobile/MessagingPushReceiver.java

```diff
@@ -0,0 +1,70 @@
+package com.adobe.marketing.mobile;
+
+import android.content.BroadcastReceiver;
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.PackageItemInfo;
+import android.content.pm.ResolveInfo;
+
+import java.util.Iterator;
+import java.util.List;
+
+public class MessagingPushReceiver extends BroadcastReceiver {
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if(context == null || intent == null) {
+            Log.warning(MessagingConstant.LOG_TAG, "onReceive() - Intent or Context null, ignoring this broadcast message");
+            return;
+        }
+        PushActionHandlingRunnable runnable = new PushActionHandlingRunnable(context, intent);
+        new Thread(runnable).start();
```

Comment:
Why do we need to create a Thread here? If app is not running and BroadCastReceiver is created, as soon as the OnReceive returns the process of app will be killed. In that case thread execution won't be completed.